apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -202,135 +221,186 @@ export function UpdateProviderPage() {
+                <Controller
+                  name={`credentials.${credential.key}`}
+                  control={control}
+                  defaultValue=""
+                  rules={{ required: `Please enter a ${credential.displayName.toLowerCase()}` }}
+                  render={({ field }) => (
+                    <IntegrationInput credential={credential} errors={errors?.credentials ?? {}} field={field} />
+                  )}
                 />
               </InputWrapper>
-            )}
-          />
-          {selectedProvider?.credentials.map((credential: IConfigCredentials) => (
-            <InputWrapper key={credential.key}>
-              <Controller
-                name={`credentials.${credential.key}`}
-                control={control}
-                defaultValue=""
-                rules={{ required: `Please enter a ${credential.displayName.toLowerCase()}` }}
-                render={({ field }) => (
-                  <IntegrationInput credential={credential} errors={errors?.credentials ?? {}} field={field} />
-                )}
-              />
-            </InputWrapper>
-          ))}
-        </CenterDiv>
-        <Group mt={16} position="apart">
-          <Center inline>
-            <Text mr={5}>Explore our</Text>
-            <Text gradient>
-              <a href={selectedProvider?.docReference} target="_blank" rel="noopener noreferrer">
-                set-up guide
-              </a>
-            </Text>
-          </Center>
-          <Button disabled={!isDirty} submit loading={isLoadingUpdate}>
-            Update
+            ))}
+          </CenterDiv>
+          <Group mt={16} position="apart">
+            <Center inline>
+              <Text mr={5}>Explore our</Text>
+              <Text gradient>
+                <a href={selectedProvider?.docReference} target="_blank" rel="noopener noreferrer">
+                  set-up guide
+                </a>
+              </Text>
+            </Center>
+            <Button disabled={!isDirty} submit loading={isLoadingUpdate}>
+              Update
+            </Button>
+          </Group>
+        </Form>
+      </SideBarWrapper>
+      <Modal
+        title={<Title size={2}>Delete {selectedProvider.name || selectedProvider.displayName} instance?</Title>}
+        opened={isModalOpened}
+        onClose={() => setModalIsOpened(false)}
+      >
+        <Text mb={30} size="lg" color={colors.B60}>
+          Deleting a provider instance will fail workflows relying on its configuration, leading to undelivered
+          notifications.
+        </Text>
+        <Group position="right">
+          <Button variant="outline" onClick={() => setModalIsOpened(false)}>
+            Cancel
+          </Button>
+          <Button
+            loading={isDeleting}
+            onClick={() => {
+              deleteIntegrationApi({
+                id: selectedProvider.integrationId,
+                name: selectedProvider.name || selectedProvider.displayName,
```

Comment:
Sending the name here as well to make sure it can be used in the notifications as well